code/modules/ruins/objects_and_mobs/necropolis_gate.dm

```diff
@@ -236,17 +240,68 @@ GLOBAL_DATUM(necropolis_gate, /obj/structure/necropolis_gate/legion_gate)
 	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
 	var/tile_key = "pristine_tile"
 	var/tile_random_sprite_max = 24
+	var/fall_on_cross = STABLE //If the tile has some sort of effect when crossed
+	var/fallen = FALSE //If the tile is unusable
 
 /obj/structure/stone_tile/Initialize(mapload)
 	. = ..()
 	icon_state = "[tile_key][rand(1, tile_random_sprite_max)]"
 
 /obj/structure/stone_tile/Destroy(force)
-	if(force)
+	if(force || fallen)
 		. = ..()
 	else
 		return QDEL_HINT_LETMELIVE
 
+/obj/structure/stone_tile/Crossed(atom/movable/AM)
+	if(fallen)
+		return
+	var/obj/item/I
+	if(istype(AM, /obj/item))
+		I = AM
+	var/mob/living/L = AM
+	if(isliving(AM))
+		L = AM
+	if(!istype(get_turf(src), /turf/open/floor/plating/lava)) //no lava to sink into
+		return
+	switch(fall_on_cross)
+		if(COLLAPSE_ON_CROSS, DESTROY_ON_CROSS)
+			if(I && I.w_class >= WEIGHT_CLASS_BULKY) //too heavy!
+				collapse()
+			if(L && !(L.movement_type & FLYING) && L.mob_size >= MOB_SIZE_HUMAN) //too big!
+				collapse()
```

Comment:
to clairify I mean `if((I && I.w_class >= WEIGHT_CLASS_BULKY) || (L && !(L.movement_type & FLYING) && L.mob_size >= MOB_SIZE_HUMAN))`